bookstore/client/bookstore.py

```diff
@@ -4,18 +4,35 @@
 
 
 class BookstoreClient(CurrentNotebookClient):
+    def __init__(self, s3_bucket=""):
+        if s3_bucket:
+            self.default_bucket = s3_bucket
+        super().__init__()
+
     @property
     def publish_endpoint(self):
         api_endpoint = "/api/bookstore/published/"
         return f"{self.url}{api_endpoint}"
 
-    def publish(self):
-        nb_json = self.get_contents(self.notebook['path'])['content']
-        req = {
-            "json": {"type": "notebook", "content": nb_json},
-            "headers": {"Content-Type": "application/json", 'Authorization': f'token {self.token}'},
-        }
+    def publish(self, path=None):
+        if path is None:
+            path = self.session.path
+        nb_json = self.get_contents(self.session.path)['content']
+        json_body = {"type": "notebook", "content": nb_json}
+
+        target_url = f"{self.publish_endpoint}{self.session.path}"
+
+        resp = self.req_session.put(target_url, json=json_body)
+        return resp
 
-        target_url = f"{self.publish_endpoint}{self.notebook['path']}"
+    @property
+    def clone_endpoint(self):
+        api_endpoint = "/api/bookstore/cloned/"
```

Comment:
cloned feels a bit off on the api to me. I would prefer a noun: clone, copy, content...